hack/tools/tilt-prepare/main.go

```diff
@@ -72,8 +75,23 @@ var (
 	kustomizeBuildsFlag  = pflag.StringSlice("kustomize-builds", []string{}, "list of kustomize build to be run; each value should be in the form name:path")
 	providersBuildsFlag  = pflag.StringSlice("providers", []string{}, "list of providers to be installed; each value should be in the form name:path[:debug]")
 	allowK8SContextsFlag = pflag.StringSlice("allow-k8s-contexts", []string{}, "Specifies that Tilt is allowed to run against the specified k8s context name; Kind is automatically allowed")
+	tiltSettingsFileFlag = pflag.String("tilt-settings-file", "", "Path to a tilt-settings.(json|yaml) file")
 )
 
+type debugConfig struct {
+	Continue     *bool `json:"continue"`
+	Port         int   `json:"port"`
+	ProfilerPort int   `json:"profiler_port"`
+	MetricsPort  int   `json:"metrics_port"`
+}
+
+type extraArgs []string
+
+type tiltSettings struct {
+	Debug     map[string]*debugConfig `json:"debug,omitempty"`
+	ExtraArgs map[string]extraArgs    `json:"extra_args,omitempty"`
```

Comment:
Additional note: I wonder if *debugConfig should be `debugConfig` we can always take the pointer of a map entry if we want to hand it over to funcs as a pointer
(I think I would do this consistently for debugConfig and extraArgs)

Reply:
AFAIK you cannot take a pointer to an element of a map in go (`cannot take the address of ...`). So if we going to set defaults in the structs inside the map those should be pointers. I'm also for consistency but a pointer to a string slice here also feels kinda weird :thinking:

Reply:
Okay, let's tackle the other things first, then I'll take a closer look what we can do here :).

It just seems strange to me that debugConfig is a pointer here, given that the map entry just shouldn't exist if there's no debugConfig for a specific provider. I get why the fields of debugConfig are pointers (and I agree) but I don't think debugConfig as map value should be a pointer.

Reply:
I changed it so that debugConfig is not a pointer anymore.

Reply:
Let's resolve for now